### ISSUE TITLE
`ShellTest.EncodeImageGivesExternalTypedData` doesn't execute for impeller

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#121623]
I think shell tests in general aren't executed for impeller.  I tried to get this test running by setting `Settings.enable_impeller`.  The encode image calls would fail then because `ShellTestPlatformView::GetImpellerContext()` returns nullptr.  I tried making `ShellTestPlatformView::GetImpellerContext()` return a mock context but `SnapshotControllerImpeller::DoMakeRasterSnapshot()` would still fail because `GetDelegate().GetSurface()` is nullptr.  I think a deeper dive will have to be done to get these tests running for impeller.

**Comment by dnfield** [comment]
On Skia, we make these use SwANGLE.

There's no software renderer for Impeller currently. I don't have a great sense of whether we'd want to use a software backend with Impeller to test things, or "just" use whatever GPU is available on the test bot assuming one is availble.

**Comment by gaaclarke** [comment]
> On Skia, we make these use SwANGLE.
> 
> There's no software renderer for Impeller currently. I don't have a great sense of whether we'd want to use a software backend with Impeller to test things, or "just" use whatever GPU is available on the test bot assuming one is availble.

Would it be possible to use Skia's software renderer with some adapter between the impeller interface and skia?

**Comment by chinmaygarde** [comment]
Can we just reword this to say "Shell unit-tests must have an Impeller backend.". I bet there are so many other things to wire up there.
